src/apis/evm/importtx.ts

```diff
@@ -229,30 +229,32 @@ export class ImportTx extends EVMBaseTx {
     }
   }
 
-  private validateApricotPhaseTwo() {
-      // enforce uniqueness of pair(address, assetid) for each out
-      let seenAssetSends: Map<string, string[]> = new Map();
-      this.outs.forEach((out: EVMOutput) => {
+  private validateApricotPhaseTwo(): void {
+      // enforce uniqueness of pair(address, assetId) for each out
+      const seenAssetSends: Map<string, string[]> = new Map();
+      this.outs.forEach((out: EVMOutput): void => {
         const address: string = out.getAddressString();
-        const assetID: string = out.getAssetID().toString('hex');
-        if (seenAssetSends.has(address)) {
-          let assetsSentToAddress: string[] = seenAssetSends.get(address);
-          if (assetsSentToAddress.includes(assetID)) {
-            let errorMessage: string = `Error - ImportTx validating Apricot Phase Two rules: duplicate (address, assetID) pair found in outputs: (${address},${assetID})`;
+        // TODO - does this need to be error detected? (capital letters)
+        const assetId: string = bintools.cb58Encode(out.getAssetID());
+        if(seenAssetSends.has(address)) {
+          const assetsSentToAddress: string[] = seenAssetSends.get(address);
+          if(assetsSentToAddress.includes(assetId)) {
+            // TODO - should the address have error detection?
+            const errorMessage: string = `Error - ImportTx validating Apricot Phase Two rules: duplicate (address, assetId) pair found in outputs: (0x${address}, ${assetId})`;
             throw new EVMOutputError(errorMessage);
           }
-          assetsSentToAddress.push(assetID);
+          assetsSentToAddress.push(assetId);
         } else {
-          seenAssetSends.set(address, [assetID]);
+          seenAssetSends.set(address, [assetId]);
         }
       });
       // make sure this transaction pays the required avax fee
-      const requiredFee: BN = Defaults[DefaultNetworkID].C.txFee; // REVIEWER QUESTION is this the appropriate way of getting the tx fee? We don't have access to the evm API here
-      let feeDiff: BN = new BN(0);
-      const AVAXAssetIDHex: string = "dbcf890f77f49b96857648b72b77f9f82937f28a68704af05da0dc12ba53f2db"; // TODO/REVIEWER QUESTION: the appropriate way of getting this ID is by querying the API, but we don't have access to the evm API here
+      const requiredFee: BN = Defaults.network[DefaultNetworkID].C.txFee;
+      const feeDiff: BN = new BN(0);
```

Comment:
same question, it's okay to manipulate `const`? I misunderstood I didn't realize `const` only meant "no reassignment"

Reply:
TLDR yes correct, `const` only means "no reassignment." 

For example check out this `ts-node` REPL session

```ts
ts-node

const map: Map<string, number> = new Map()
map.set('foo', 123)

map
Map(1) { 'foo' => 123 }

map.has('foo')
true

map.delete('foo')
true

map.has('foo')
false

map.set('bar', 123)
Map(1) { 'bar' => 123 }

map.clear()
map
Map(0) {}
```

Reply:
I'm able to manipulate `map` even though it's a `const`.